src/shared/kbd-util.c

```diff
@@ -145,11 +145,11 @@ int keymap_exists(const char *name) {
                                 &(struct recurse_dir_userdata) {
                                         .keymap_name = name,
                                 });
-                if (r == -ENOENT)
-                        continue;
-                if (ERRNO_IS_RESOURCE(r))
-                        return r;
                 if (r < 0) {
+                        if (r == -ENOENT)
+                                continue;
+                        if (ERRNO_IS_RESOURCE(r))
```

Comment:
Why not just make this `if (r < 0 && ERRNO_IS_RESOURCE(r))` ? Then it's a one-line-change and we don't get the extra indentation layer.

Reply:
Well, the `if (r < 0)` branch is already there, so it's only logical to move the `ERRNO_IS_RESOURCE(r)` check where it belongs, isn't it?

Reply:
Note that before this series most of the code was using `ERRNO_IS_*(r)` correctly, and the `(r < 0 && ERRNO_IS_*(r))` format was used just in 20% cases, remaining `ERRNO_IS_*(r)` were in `(r < 0)` branches is already.

Reply:
> Well, the `if (r < 0)` branch is already there, so it's only logical to move the `ERRNO_IS_RESOURCE(r)` check where it belongs, isn't it?

I agree that this makes more sense.

Reply:
I like the change here, and most of changes in this PR.

Reply:
"Where it belongs" is subjective. The compiler is perfectly capable of emitting the same code if the code is written in a way where a condition is evaluated twice. So the primary motivation should be for the code to be easy to read and easy to edit. With the flat structure, both things are much easier. We're writing the code for people, not for the compiler.

Reply:
My point is that it is a bit easier to read the code the way I wrote it.  :)
One could argue that when the indentation is already too deep,
one level more doesn't necessarily makes things easier to read.
Most of the changes in this patchset are not of that kind, though.